src/main/java/com/team766/controllers/ExtendableCanSparkMaxSmartMotionPIDController.java

```diff
@@ -0,0 +1,283 @@
+package com.team766.controllers;
+
+import com.team766.config.ConfigFileReader;
+import com.team766.hal.RobotProvider;
+import com.team766.library.SetValueProvider;
+import com.team766.library.SettableValueProvider;
+import com.team766.library.ValueProvider;
+import com.team766.logging.Logger;
+import com.team766.logging.LoggerExceptionUtils;
+import edu.wpi.first.math.MathUtil;
+import com.revrobotics.SparkMaxAbsoluteEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.SparkMaxAbsoluteEncoder.Type;
+import java.io.IOError;
+import javax.swing.text.DefaultStyledDocument.ElementSpec;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.revrobotics.CANSparkMax;
+import com.team766.framework.Mechanism;
+import com.team766.hal.MotorController;
+import com.team766.library.RateLimiter;
+import com.team766.controllers.PIDSTATE.*;
+import com.team766.framework.Exceptions.*;
+
+
+public abstract class ExtendableCanSparkMaxSmartMotionPIDController {
```

Comment:
usually we name it `Base` or `Abstract`
e.g. `AbstractCanSparkMaxSmartMotionPIDController`